### PR TITLE
build: fix symlinks in lv2dir

### PR DESCRIPTION
**Pull request by puleglot**
Fixes loading of lv2 plugin and its gui.

### CODE REVIEW

**Comment by falkTX** [comment]
Which ticket/issue does this fix?
It has been working so far as-is for many people, I think applying these changes will break existing setups.

**Comment by puleglot** [comment]
This should fix regression introduced by https://github.com/calf-studio-gear/calf/commit/d18a873ebc92bd1eee2b3f61b7768a195fe6906a and https://github.com/calf-studio-gear/calf/commit/75f33e02d3c9d0acf2bee59fd286cbc98f0d3d67

**Comment by puleglot** [comment]
Downstream bug: https://bugs.gentoo.org/779343

**Comment by gitterdude** [comment]
The last working git version in Ubuntu 18.04 right now is 4e699444; just prior to d18a873e
The newer ones fail as stated in the gentoo report.

File: src/Makefile.am [hunk 1]
```diff
@@ -79,7 +79,8 @@ if USE_GUI
 endif
 if USE_LV2
 	install -d -m 755 $(DESTDIR)$(lv2dir)
-	ln -sf $(pkglibdir)/calf.so $(DESTDIR)$(lv2dir)/calf.so
+	ln -sf $(pkglibdir)/libcalf.so $(DESTDIR)$(lv2dir)/calf.so
+	ln -sf $(pkglibdir)/libcalflv2gui.so $(DESTDIR)$(lv2dir)/calflv2gui.so
```
Comment (falkTX) [inline src/Makefile.am:83]: this line needs a check for `if USE_LV2_GUI`

**Comment by falkTX** [comment]
So seems fine to me, a regression from a recent commit.
I added a little comment about something that needs attention, rest seems fine.
Thanks

**Comment by boomshop** [comment]
Great, thanks a bunch for taking care of this one @falkTX and @puleglot 

Please let me know when this fix is in a state able to be merged and I'll pull the trigger on it.

**Comment by falkTX** [comment]
The changes look good
Though I did not verify if the build is indeed fixed (would be great to have some CI/automated testing..)

**Comment by nick87720z** [comment]
Probably this #294

**Comment by puleglot** [comment]
> But pulseeffects still cries.

What problem with pulseeffects? I tested with Ardour only, but in theory calf should be loaded fine by any of LV2 host software.

**Comment by zettberlin** [comment]
> I tried to edit symlink to point to libcalf.so. This made it loadable in carla, but without special gui.

Same here. 
I created a symlink from libcalf.so to calf.so. Plugins load and can be edited with a generic host UI but their own GUI does not load. No entry in Carlas log and none in Ardours log either...

**Comment by zettberlin** [comment]
Tried again with this branch, to no avail...
Filenames are as wrong as before in /usr/local/lib/calf (no calf.so but libcalf.so) and Carla and Ardour complain.

What do I do wrong?

**Comment by zettberlin** [comment]
I now set a symlink from python to python3, it does look a little different in Carla, as in, now I get something in the logs:

Will use UI-Bridge for 'GxAmplifier-Stereo-X', binary: "/usr/lib/carla/carla-bridge-lv2-x11"
ALL OK!
Will use UI-Bridge for 'Calf Flanger', binary: "/usr/lib/carla/carla-bridge-lv2-gtk2"
ALL OK!
Carla assertion failure: "fLibFilename.isNotEmpty()" in file CarlaBridgeFormat.cpp, line 104
Failed to load UI binary, error was:
(null)

As you see: the Guitarix GUI loads OK, Calf Flanger does not...

**Comment by kisaragi-hiu** [comment]
Arch Linux, but this should also replicate everywhere else:

- `git clone https://github.com/calf-studio-gear/calf calf`
- `cd calf`
- Remove the existing Calf installation, if any
- `./autogen.sh`
- `make`
- `sudo make install`
- Open Ardour and insert Calf Compressor
- Observe the error --- Ardour's log only shows one line, that it failed to instantiate
- Open REAPER, for that matter, and insert Calf Compressor
- Observe even the generic UI fails to load

This patch seems to be necessary to get `master` builds to load again.

---

> Filenames are as wrong as before in /usr/local/lib/calf (no calf.so but libcalf.so)

`.../lib/calf/libcalf.so` is the correct name. `.../lib/lv2/calf.lv2/calf.so` being a symlink to inexistant `.../lib/calf/calf.so` is the incorrect part.

> I tried to edit symlink to point to libcalf.so
> I created a symlink from libcalf.so to calf.so

Creating symlinks for libraries yourself is generally not a reliable way to get things to load properly.

---

Would it help if GitHub Actions is set up to at least check that `./autogen.sh; make` succeeds?

**Comment by JohannesLorenz** [comment]
As requested, I was able to run this PR in CI (on a temporary branch) with success for linux-gcc, linux-clang, macos and mingw64: https://github.com/calf-studio-gear/calf/actions/runs/6578998596 .

If I understand the comments right, this is safe to merge now?

**Comment by JohannesLorenz** [comment]
Friendly reminder that I have an open question in the above post. Would be nice if anyone could help here.

**Comment by kisaragi-hiu** [comment]
Thank you for the maintenance; I'll take a look tomorrow to see how Calf builds on my machine (now using KDE Neon) before and after this patch.

**Comment by kisaragi-hiu** [comment]
Thank you for your maintenance; I'll take a look tomorrow to see how Calf builds and loads on my machine before and after this patch. (I remembered I want to help after seeing the ping, but I'm currently away from my PC.)

**Comment by kisaragi-hiu** [comment]
I've tested again and the results still agree with my previous result, where the plugin does not load without this PR, and does with it applied.

With current master (4c0d80568a748f32dc4368150c30fa267f711a36):

```sh
git clone https://github.com/calf-studio-gear/calf calf
cd calf

./autogen.sh
./configure
make # assuming lv2 and other dependencies are already present
sudo make install
```

Then open a DAW and try to load a Calf plugin. It fails to load.

With this PR applied:

```sh
# cleanup
sudo make uninstall
git clean -Xdf
# apply this PR onto the just-cloned copy
gh pr checkout 290 # using github cli
git rebase origin/master # just fast forward for the test

./autogen.sh
./configure
make # assuming lv2 and other dependencies are already present
sudo make install
```

Now open eg. Ardour and try to load a Calf plugin. It loads successfully.

**Comment by JohannesLorenz** [comment]
@kisaragi-hiu Thanks a lot! I guess we can just merge this PR then?

**Comment by JohannesLorenz** [comment]
Will merge in 3 days if no one will disagree.

**Comment by JohannesLorenz** [comment]
While this patch provided correct *relative* symlinks, I will now convert them to *absolute* due to the discussion in #306 (especially https://github.com/calf-studio-gear/calf/issues/306#issuecomment-1902843108, based on this [FreeBSD port patch](https://cgit.freebsd.org/ports/tree/audio/calf-lv2/files/patch-src_Makefile.am)).

I mention this here because there was a lot of discussion about those symlinks here. If anyone disagrees with absolute links, please let me know here (or in #306). Otherwise, I will convert the links to absolute links in the next days.

**Pull request merged**